api/v1beta1/common_types.go

```diff
@@ -148,6 +148,26 @@ const (
 	// This annotation can be used to inform MachinePool status during in-progress scaling scenarios.
 	ReplicasManagedByAnnotation = "cluster.x-k8s.io/replicas-managed-by"
 
+	// AutoscalerMinSizeAnnotation defines the minimum node group size.
+	// The annotation is used by autoscaler.
+	// The annotation is copied from kubernetes/autoscaler.
+	// Ref:https://github.com/kubernetes/autoscaler/blob/d8336cca37dbfa5d1cb7b7e453bd511172d6e5e7/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go#L256-L259
+	// Note: With the Kubernetes autoscaler it is possible to use different annotations by configuring a different
+	// "Cluster API group" than "cluster.x-k8s.io" via the "CAPI_GROUP" environment variable.
+	// We only handle the default group in our implementation.
+	// Note: It can be used by setting as top level annotation on MachineDeployment and MachineSets.
+	AutoscalerMinSizeAnnotation = "cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size"
```

Comment:
I would prefer to not expose auto scaler annotations as part of the CAPI API, even if this means duplicating the const in the test framework for now (might be in the future we define them somewhere in a shared internal package)

Reply:
It was your suggestion 😀 https://github.com/kubernetes-sigs/cluster-api/pull/8514#discussion_r1164020765

Reply:
already discussed in https://github.com/kubernetes-sigs/cluster-api/pull/8514#discussion_r1164020765

Reply:
I'm also fine with keeping it internal